server/tserver/src/main/java/org/apache/accumulo/tserver/log/TabletServerLogger.java

```diff
@@ -458,6 +460,27 @@ private void write(final Collection<CommitSession> sessions, boolean mincFinish,
       // the logs haven't changed.
       final int finalCurrent = currentLogId;
       if (!success) {
+        final ServiceLock tabletServerLock = tserver.getLock();
+        if (sawWriteFailure && tabletServerLock != null) {
+          log.info("WAL write failure, validating server lock in ZooKeeper");
+          if (!tabletServerLock.verifyLockAtSource()) {
+            // try to close the log, then Halt the VM
+            testLockAndRun(logIdLock, new TestCallWithWriteLock() {
+
+              @Override
+              boolean test() {
+                return true;
+              }
+
+              @Override
+              void withWriteLock() throws IOException {}
+            });
+
+            log.error("Writing to WAL has failed and TabletServer lock does not exist. Halting...");
```

Comment:
The halt code will attempt to log an error message so do not have to log anything here.   Can the testLockAndRun code be removed now that its not closing the file?

```suggestion
```

Reply:
I'm not sure which is why I wanted you to review.

Reply:
I see what you are saying, this isn't doing anything now, so can be removed. I can remove this.

Reply:
5b95b50